src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java

```diff
@@ -14,4 +30,59 @@ public String getRecoveredAdUnitId(String adUnitId) {
     jsActions.waitForJavaScriptTruthy(String.format("window._sp_ && window._sp_.getElementId && %s", getIdScript));
     return jsActions.execute(getIdScript).toString();
   }
+
+  public void verifyPageFairRecoveryWithAdBlock() {
+    int recoverableAdsCount = 2;
+    By spansBodyChildrenSelector = By.cssSelector("body>span");
+    Dimension topLeaderboardSize = new Dimension(728, 90);
+    Dimension medrecSize = new Dimension(300, 250);
+    String expectedRecoveredLB;
+    String expectedRecoveredMR;
+
+    try {
+      expectedRecoveredLB = readFileToString(new File(EXPECTED_TOP_LEADERBOARD_PATH));
+      expectedRecoveredMR = readFileToString(new File(EXPECTED_MEDREC_PATH));
+    } catch (IOException e) {
+      PageObjectLogging.log("Can't open expected PageFair recovery file.", e, false);
+      throw new WebDriverException("Can't open expected PageFair recovery file.");
+    }
+
+    // when PF recovered ad is on page, inserts span elements as a direct children of body
+    wait.forElementPresent(spansBodyChildrenSelector);
+
+    // verify that adblock is turned on on that page
+    verifyNoAdsOnPage();
+
+    String firstSpanClass = driver.findElement(spansBodyChildrenSelector).getAttribute("class");
+    List<WebElement> recoveredAds = driver
+        .findElements(By.cssSelector("body>span." + firstSpanClass))
+        .stream()
+        .filter(WebElement::isDisplayed)
+        .filter(e -> e.getCssValue("background").contains("data:image/jpeg"))
+        .collect(Collectors.toList());
+
+    Assert.assertEquals(recoveredAds.size(), recoverableAdsCount);
+
+    for (WebElement ad : recoveredAds) {
+      Dimension adSize = ad.getSize();
+
+      if (adSize.equals(topLeaderboardSize)) {
+        Assertion.assertTrue(ad.getCssValue("background").contains(expectedRecoveredLB), "TOP_LEADERBOARD is not correctly recovered!");
+      } else if (adSize.equals(medrecSize)) {
+        Assertion.assertTrue(ad.getCssValue("background").contains(expectedRecoveredMR), "MEDREC is not correctly recovered!");
+      } else {
+        Assertion.fail("Not supported PageFair recovery ad size encountered: " + adSize);
+      }
+    }
+  }
+
+  public void verifyPageFairRecoveryWithNoAdBlock() {
+    verifyTopLeaderboard();
+    verifyMedrec();
+
+    // TODO: uncomment after ADEN-5041 is merged
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)